improve unit test code coverage for storage.go to 77%

Bee-lee opened the pull request:
# Description
only missing function in storage.go is about initializing a database

overall percentage 26% -> 50%

Fixes 

## Type of change

Please delete options that are not relevant.

- Unit tests (no changes in the code)

## Testing steps

Please describe how the change was tested locally. If, for some reason, the testing was not done or not done fully, please describe what are the testing steps.

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

tisnik commented on storage.go:
```diff
@@ -106,7 +106,7 @@ func initDatabaseConnection(configuration StorageConfiguration) (*sql.DB, error)
 func displayMultipleRuleDisable(connection *sql.DB, output string) error {
 	var fout *os.File = nil
-	var writer *bufio.Writer = nil
+	writer := bufio.NewWriter(os.Stdout)
```
it does not seem to be correct - it will output to stdout which is not always needed. + the defer might not work as well as expected.

Bee-lee replied:
done

tisnik commented on storage.go:
```diff
@@ -265,7 +264,7 @@ func readOrgID(connection *sql.DB, clusterName string) (int, error) {
 func displayAllOldRecords(connection *sql.DB, maxAge, output string) error {
 	var fout *os.File = nil
-	var writer *bufio.Writer = nil
+	writer := bufio.NewWriter(os.Stdout)
```
dtto

Bee-lee replied:
done

tisnik commented on storage_test.go:
```diff
@@ -241,6 +242,179 @@ func TestPerformDisplayMultipleRuleDisableOnError(t *testing.T) {
+`
+	// call the tested function
+	err = cleaner.PerformDisplayMultipleRuleDisable(connection, nil, query1, "cluster_rule_toggle")
+	if err != nil {
```
`assert.Error` or `assert.NoError` is better

tisnik replied:
the tricky part - check if the function does it's job, i.e. what is displayed actually?

Bee-lee replied:
done both

tisnik commented on storage_test.go:
```diff
@@ -241,6 +242,179 @@ func TestPerformDisplayMultipleRuleDisableOnError(t *testing.T) {
+	}
+
+	err = connection.Close()
+	if err != nil {
```
dtto with assertion

tisnik commented on storage_test.go:
```diff
@@ -435,3 +685,134 @@ func TestDeleteRecordFromTableOnError(t *testing.T) {
+
+	mock.ExpectClose()
+
+	_, err = cleaner.PerformCleanupInDB(connection, clusterNames)
```
todo -check for actual return value?

Bee-lee replied:
done

tisnik reviewed (commented):
It looks ok, just added couple of comment/ideas

tisnik reviewed (changes requested):
Please add tests for negative cases too. It might improve coverage a bit :)

Bee-lee commented:
@tisnik added negative test cases for scan errors, added check for output from `PerformCleanupInDB`, added checks for generated files from `DisplayAllOldRecords` and `DisplayMultipleRuleDisable`, added asserts instead of err checks, DRYed up some stuff

tisnik reviewed (approved):
LGTM
